src/main/dsl/AnyOf.kt

```diff
@@ -0,0 +1,71 @@
+package dsl
+
+class AnyOf(private val content : ArrayList<Terminal>) : Group() {
+    
+    constructor(s : String) : this(parseString(s))
+
+//    Literals can be sequences of multiple characters and must be handled differently
+    override fun render(builder: StringBuilder) {
+        var previousIsLiteral = false
+    
+        for (e in content) {
+            if(e is Literal){
+                if(!previousIsLiteral) {
+                    builder.append('(')
+                } else {
+                    builder.append('|')
+                }
+                builder.append('"')
+                
+                e.render(builder)
+                
+                builder.append('"')
+                previousIsLiteral = true
+            } else {
+                if(previousIsLiteral){
+                    builder.append(")|[")
+                } else if (e == content[0]) {
+                    builder.append("[")
+                }
+                e.render(builder)
+            }
+            
+        }
+    
+        if(previousIsLiteral){
+            builder.append(")")
+        } else {
+            builder.append("]")
+        }
+        
+    }
+    
+    
+    fun literal(init: () -> String): Literal {
+        val lit = Literal(init())
+        content.add(lit)
+        return lit
+    }
+    
+}
+
+class Enum(val v : String) : Terminal() {
```

Comment:
We might want to consider renaming this to prevent conflicts with the built in Enum type.